clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -79,6 +81,25 @@ public KafkaLZ4BlockInputStream(InputStream in) throws IOException {
     }
 
     /**
+     * Create a new {@link InputStream} that will decompress data using the LZ4 algorithm.
+     *
+     * @param in The stream to decompress
+     * @throws IOException
+     */
+    public KafkaLZ4BlockInputStream(InputStream in) throws IOException {
```

Comment:
I would really hate to remove this. It totally sucks that kafka can't use the simple interface right now because it has to maintain backwards compatibility. But I hope that others using these classes that are not constrained by compatibility issues will be able to use the easy / simple / correct constructor.